simplicity-sys/src/dag.rs

```diff
@@ -0,0 +1,120 @@
+// Simplicity Bindings
+// Written in 2022 by
+//   Christian Lewe <clewe@blockstream.com>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+use crate::bitstream::Bitstream;
+use crate::bitstring::Bitstring;
+use crate::error::Error;
+use libc::{c_void, size_t};
+use std::ops::Index;
+
+/// Kind of Simplicity node.
+///
+/// Corresponds to `tag_t` in `depend/simplicity/dag.h`.
+#[repr(C)]
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub enum Tag {
+    COMP,
+    CASE,
+    ASSERTL,
+    ASSERTR,
+    PAIR,
+    DISCONNECT,
+    INJL,
+    INJR,
+    TAKE,
+    DROP,
+    IDEN,
+    UNIT,
+    HIDDEN,
+    WITNESS,
+    JET,
+}
+
+/// (First node of) a Simplicity DAG (typed or untyped).
+///
+/// Corresponds to `dag_node` in `depend/simplicity/dag.h`.
+#[repr(C)]
+pub struct DagNode {
+    jet: *const c_void,
+    cmr: [u32; 8],
+    // Assume largest union member
+    union_aux_types: [size_t; 2],
```

Comment:
In 8e45d76bc0e528aed678978f5111b76d2b23063a:

This makes me a little nervous. Rust has a `union` keyword. Could you use that?

Reply:
`DagNode` now uses unions, namely `AuxTypes` [[1]](https://github.com/chlewe/rust-simplicity/blob/simplicity-sys/simplicity-sys/src/dag.rs#L46) and `IndicesChildrenWitness` [[2]](https://github.com/chlewe/rust-simplicity/blob/simplicity-sys/simplicity-sys/src/dag.rs#L54).